src/lib/bit/bit.h

```diff
@@ -632,14 +636,25 @@ bit_clz_u64(uint64_t x)
  * @cond false
  * @brief Naive implementation of popcount.
  */
-#define POPCOUNT_NAIVE(x, bitsize)  {					\
-	int r;								\
-	for (r = 0; x; r++) {						\
-		x &= (x-1);						\
-	}								\
-									\
-	return r;							\
-}
+#define POPCOUNT_NAIVE(x, bitsize)  ({	\
+	int r;	\
+	for (r = 0; x; r++) {	\
+		x &= (x - 1);	\
+	}	\
+	r;	\
+})
+
+#if	defined(HAVE_BUILTIN_POPCOUNTLL)
+#	define POPCOUNT64 __builtin_popcountll
+#	define POPCOUNT32 __builtin_popcount
+#else	/* !defined(HAVE_BUILTIN_POPCOUNT) */
+#	define POPCOUNT64(x)  POPCOUNT_NAIVE((x), sizeof(uint64_t) * CHAR_BIT)
+#	define POPCOUNT32(x)  POPCOUNT_NAIVE((x), sizeof(uint32_t) * CHAR_BIT)
+#endif
+
+#define POPCOUNT(x) (sizeof((x)) == 32	\
+				? POPCOUNT32((x))	\
+				: POPCOUNT64((x)))
```

Comment:
I don't understand why you need to do that. `bit_count_u32` and `bit_count_u64` seem fine as they are.